ENG-1345 selection_tag docs (#130)
NHSDigital/terraform-aws-backup: README.md

When the earlier step wrote to the `$GITHUB_ENV` file, it wrote the `destination_vault_arn` output from the destination configuration. This is passed in as the `destination_vault_arn` variable to the source configuration, so we don't need to explicitly configure it here. Your CI/CD pipeline will need to ensure that the destination configuration is applied before the source configuration and that the `destination_vault_arn` output is passed along, and probably has a similar mechanism to the above.

## Tagging Strategy

The `selection_tag` variable controls which resources AWS Backup will protect. It is fully configurable — you can use any tag key and value that suits your project. The examples in this README use `NHSE-Enable-Backup` as the tag key with a value of `True`, but this is a recommendation, not a requirement.

### Choosing a selection tag

For **production** environments, we recommend using `NHSE-Enable-Backup` = `True` as the standard tag. This aligns with organisational reporting and makes it straightforward for centralised compliance dashboards to identify backed-up resources.

For **non-production** environments, you have flexibility:

> - Use the same tag (`NHSE-Enable-Backup` = `True`) if you want identical backup behaviour and are happy for the same plan to cover all tagged resources in the account.

Reviewer: ```suggestion
- Use the same tag (`NHSE-Enable-Backup` = `True`) if you are going to create a single backup plan in the account.
```

> - Use a different tag value (e.g. `NHSE-Enable-Backup` = `dev`) or a different tag key entirely (e.g. `BackupDev` = `True`) if you need to distinguish which resources belong to which environment's backup plan.

Reviewer: ```suggestion
- Use a different tag values (e.g. `NHSE-Enable-Backup` = `dev`) or a different tag keys entirely (e.g. `BackupDev` = `True`) if you will have multiple backup plans in the account and need to distinguish which resources belong to which backup plan.
```

Author: updated, chose slightly different wording but added text for multiple
